### ISSUE TITLE
Error in go#fmt#update_file

### ISSUE THREAD

**Issue by Contextualist** [fatih/vim-go#1409]
### Behavior

Each time go#fmt#update_file is called (e.g. saving a file), an error message shows up.
```
Error detected while processing function <SNR>71_fmt_autosave[3]..go#fmt#Format[45]..go#fmt#update_file:
line   24: 
E118: Too many arguments for function: getqflist
E15: Invalid expression: getqflist({'title': 1})
```

p.s. seem to relate to changes in #1407 .

### Steps to reproduce:

1. Open a .go file
2. :w

### Configuration

* vim version: 7.4.1289
* vim-go version: a9ad6b3
* go version: 1.9rc2

**Comment by fatih** [comment]
Thanks for the feedback. Seems like it was introduced with Vim 8. If you upgrade your Vim it should be fixed, nevertheless let me add a check so it doesn't cause problems for others.

**Comment by themue** [comment]
Same here.

- vim version 7.4.1289
- go version 1.8.3
- vim-go version `df99b19`

**Comment by flowerinthenight** [comment]
Same here as well. Updated plugins this morning, did a `GoUpdateBinaries`.

```
Error detected while processing function <SNR>40_fmt_autosave[3]..go#fmt#Format[45]..go#fmt#update_file:
line   24:
E118: Too many arguments for function: getqflist
E15: Invalid expression: getqflist({'title': 1})
line   25:
E121: Undefined variable: l:list_title
E116: Invalid arguments for function has_key(l:list_title, "title") && l:list_title['title'] == "Format"
```

**Comment by fatih** [comment]
Thanks for the feedback. This is now fixed in master.

**Comment by vhakulinen** [comment]
I had this problem with neovim 0.1.7 but managed to fix by updating to neovim 0.2.0 (did this before that patch).

Just dropping my two cents here if someone has the same problem with neovim.
